File: backend/app/sensors/registry.py
from __future__ import annotations

from app.sensors.base import BaseSensorSimulator
from app.sensors.em_induction import EMInductionSimulator
from app.sensors.gpr import GPRSimulator
from app.sensors.gravimetric import GravimetricSimulator
from app.sensors.lidar import LidarSimulator
from app.sensors.magnetometer import MagnetometerSimulator
from app.sensors.resistivity import ResistivitySimulator
from app.sensors.seismic import SeismicSimulator
from app.sensors.thermal import ThermalSimulator

SENSOR_REGISTRY: dict[str, BaseSensorSimulator] = {
    s.key: s
    for s in [
        GPRSimulator(),
        MagnetometerSimulator(),
        ResistivitySimulator(),
        SeismicSimulator(),
        EMInductionSimulator(),
        ThermalSimulator(),
        GravimetricSimulator(),
        LidarSimulator(),
    ]
}

ALL_SENSOR_KEYS = list(SENSOR_REGISTRY.keys())

File: frontend/src/sim/sensors/registry.ts
import type { SensorDef } from "./base";
import { GPRSensor } from "./gpr";
import { MagnetometerSensor } from "./magnetometer";
import { ResistivitySensor } from "./resistivity";
import { SeismicSensor } from "./seismic";
import { EMInductionSensor } from "./emInduction";
import { ThermalSensor } from "./thermal";
import { GravimetricSensor } from "./gravimetric";
import { LidarSensor } from "./lidar";

export const SENSOR_REGISTRY: Record<string, SensorDef> = {
  gpr: GPRSensor,
  magnetometer: MagnetometerSensor,
  resistivity: ResistivitySensor,
  seismic: SeismicSensor,
  em_induction: EMInductionSensor,
  thermal: ThermalSensor,
  gravimetric: GravimetricSensor,
  lidar: LidarSensor,
};

export const ALL_SENSOR_KEYS = Object.keys(SENSOR_REGISTRY);
